packages/expo-network/android/src/main/java/expo/modules/network/NetworkModule.java

```diff
@@ -120,6 +120,19 @@ private NetworkStateType getNetworkCapabilities(NetworkCapabilities nc) {
     return NetworkStateType.UNKNOWN;
   }
 
+  private String rawIpToString(Integer ip) {
+    // Convert little-endian to big-endian if needed
+    if (ByteOrder.nativeOrder().equals(ByteOrder.LITTLE_ENDIAN)) {
+      ip = Integer.reverseBytes(ip);
+    }
+    byte[] ipByteArray = BigInteger.valueOf(ip).toByteArray();
+    try {
+      return InetAddress.getByAddress(ipByteArray).getHostAddress();
+    } catch (UnknownHostException e) {
```

Comment:
This exception is thrown if the byte array is of the wrong length. While I think it's good to have this `catch` clause if we document this behavior in our API contract, this also means that IP addresses that start with a zero but aren't "0.0.0.0" (uncommon but valid within a network) will all get forced into this code path.

The byte array needs to be front-padded with zero bytes so that it always is four bytes long.

Reply:
hm, I don't understand- why would an ip address that just starts with zero result in an array of the wrong length? 

I tried triggered the error by providing the `rawIpToString` method with the integer `16885760`, and got back `0.168.1.1` (as expected).

Reply:
If it starts with a zero the integer will be too small to produce a four byte array. Try it.

Reply:
ahh I see now, thanks for pointing that out